### PR TITLE
Day 3/milena

### PR DESCRIPTION
**Pull request by milena-kolosek**

### CODE REVIEW

**Comment by kristina-kolosek** [comment]
I get an error "Cannot read property 'status' of undefined" when trying to save api key (src/reducers/AuthenticationReducer/actions.js:18)

File: src/CircleCiRoute/index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { Route, Redirect } from 'react-router-dom';
+
+const CircleCiRoute = ({ component: Component, ...rest }) => (
+  <Route {...rest} render={(props) => (
+    sessionStorage.getItem('circleCiToken') ? <Component {...props} /> : <Redirect to='/auth' />
```
Comment (kristina-kolosek) [inline src/CircleCiRoute/index.js:6]: Great job on this, this is the most proper way to handle authorization

File: src/components/CircleCiBuildDetails/index.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+          <label>Build no #{buildNo}</label>
+        </div>
+        <div className='details-data'>
+          Project name: {projectName}
+        </div>
+        <div className='details-data'>
+          Branch name: {branchName}
+        </div>
+        <div className='details-data'>
+          Status: <label className={'status-' + status}>{status.charAt(0).toUpperCase() + status.slice(1)}</label>
```
Comment (kristina-kolosek) [inline src/components/CircleCiBuildDetails/index.js]: Either move this out as capitalize utility function, or use loads utility :). + it'd be the most proper capitalization if the rest was lower-cased

File: src/components/CircleCiBuilds/index.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      <div className='circle-ci-builds-page'>
+        <h2>Circle Ci</h2>
+        <div>
+          <label>Latest builds</label>
+        </div>
+        <div className='latest-builds'>
+          { 
+            this.props.latestBuilds.map((build, i) => 
+              <Link 
+                key={"commit-" + i}
```
Comment (kristina-kolosek) [inline src/components/CircleCiBuilds/index.js]: the most proper key for this list would be what uniquely identifies the build:

vcs type + project + branch + build number

File: src/components/Commit/index.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      <div className="commit" onClick={this.viewDetails}>
+        <div className='flex-holder-space-between'>
+          <div className='branch-details'><b>{branchName} / {projectName} #{this.props.buildNo}</b></div>
+          <div className='time'> time ago</div>
+        </div>
+        <div className='flex-holder-space-between'>
+          <div className='msg'>{ commitMsg }</div>
+          <div className='flex-holder-space-between status-holder'>
+             <label className='status'>Status: </label>
+             <label className={'status-' + status}>{status.charAt(0).toUpperCase() + status.slice(1)}</label>
```
Comment (kristina-kolosek) [inline src/components/Commit/index.js]: Same comment as above, regarding capitalization

File: src/pages/AuthenticatePage/index.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+  saveKey = (e) => {
+    this.setState({ key: e.target.value });
+  };
+
+  authenticate = (e) => {
+    e.preventDefault();
+    this.props.login(this.state.key).then(() => {
+      if (!this.props.error) {
+        this.props.history.push("/");
```
Comment (kristina-kolosek) [inline src/pages/AuthenticatePage/index.js]: better use setting of state + redirect component, a more proper way than programmatically pushing

**Review by kristina-kolosek — CHANGES_REQUESTED** [review]
Besides these comments I left, idea is to also keep the selected branch if route is refreshed In your case, it'd be a bit more complex, so try doing it for desktop devices only

**Comment by milena-kolosek** [comment]
I'm not sure what you meant by 'to keep the selected branch if route is refreshed In your case'.
I'm saving the selected branch in the store state as 'commitDetails'.

File: src/components/CircleCiBuilds/index.js [hunk 6]
```diff
@@ -31,7 +31,7 @@ class CircleCiBuilds extends Component {
           { 
             this.props.latestBuilds.map((build, i) => 
               <Link 
-                key={"commit-" + i} 
+                key={i + '-' + build.id + '-' + build.branchName + '-' + build.projectName + '-'}
```
Comment (kristina-kolosek) [inline src/components/CircleCiBuilds/index.js]: Isn't it easier to use `${i}-${build.id}-${build.branchName}-${build.projectName}-` instead? (I'm not quite a fan of string concatenation :D)

**Comment by kristina-kolosek** [comment]
Reducer state is wiped off on page refresh, so you'd need a parameter in route to persist it and make more SEO friendly

**Pull request closed without merging**